api/v1alpha1/etcdcluster_types.go

```diff
@@ -74,6 +78,11 @@ type EtcdCluster struct {
 	Status EtcdClusterStatus `json:"status,omitempty"`
 }
 
+// CalculateQuorumSize returns minimum quorum size for current number of replicas
+func (r *EtcdCluster) CalculateQuorumSize() int {
+	return int(*r.Spec.Replicas)/2 + 1
```

Comment:
It can nil pointer error. For type safety I guess we need to check replicas for nil firstly